python/setup_cluster/create_nfs.py

```diff
@@ -0,0 +1,129 @@
+from ssh_client_handler import SSHClientHandler
+
+
+class NFSSetupManager:
+    def __init__(self, config, logger):
+        self.config = config
+        self.logger = logger
+
+        nodes = self.config["nodes"]
+        self.node_ssh_username = self.config["ssh_username"]
+
+        shared_node = next(node for node in nodes if "shared_file_system" in node)
+        web_ingress_node = next(node for node in nodes if "web_ingress" in node)
+
+        self.shared_file_system_private_ip = shared_node["private_ip"]
+
+        self.web_ingress_public_ip = web_ingress_node["web_ingress"]["public_ip"]
+        self.web_ingress_private_ip = web_ingress_node["private_ip"]
+        self.web_ingress_ssh_username = web_ingress_node["web_ingress"]["ssh_username"]
+
+        self.shared_dir = shared_node["shared_file_system"]["shared_dir"]
+
+        self.nfs_client_private_ips = [
+            node["private_ip"]
+            for node in self.config["nodes"]
+            if node["private_ip"] != self.shared_file_system_private_ip
+        ]
+        self.ssh_client_handler = SSHClientHandler(
+            self.node_ssh_username,
+            self.web_ingress_ssh_username,
+            self.web_ingress_public_ip,
+            self.web_ingress_private_ip,
+            logger=logger,
+        )
```

Comment:
There is a lot of repetition of code in this part, for initializing look for cleaning all of it up.